### PR TITLE
:memo: Study of different architectures of the engine (EPI-19)

### PR DESCRIPTION
**Pull request by landryarki**
comparing ECS and OOP based game engine

### CODE REVIEW

**Review by axel-eck — CHANGES_REQUESTED** [review]
It would be nice if you can add or update the final sentence to explain our final choice (here OOP) and why we made that choice.

**Review by 6im0n — COMMENTED** [review]
Good benchmark ! Just add a litle conclusion to introduce the OOP, and the schematics of the architecture :ok_hand:

File: documentation/benchmarks/game_engine/architecture_study.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+- **OOP** encourages data encapsulation, while **ECS** promotes exposed plain old data (POD) objects.
+- **OOP** considers inheritance a first-class citizen, while **ECS** considers composition a first-class citizen.
+- **OOP** colocates data with behavior, while **ECS** separates the data from behavior.
+
+---
+
+## What should I use for a video game?
+In video game development, ECS is considered more performant than OOP, especially when managing numerous entities or when entities need to change behavior dynamically. OOP works well for traditional applications with clear hierarchies, but ECS offers flexibility and scalability in gaming.
+
+Source: [GameDev Stack Exchange](https://gamedev.stackexchange.com/questions/200076/i-dont-get-why-ecs-is-considered-more-performant-than-oop)
```
Comment (6im0n) [inline documentation/benchmarks/game_engine/architecture_study.md]:
```suggestion

The flowchart of our architecture 
```mermaid
flowchart LR

    scene_manager[scene manager]
    event_sys[event system]
    object_management[object manager]
    config_reader[config reader]
    factory[factory]
    
    subgraph Game_engine
        scene_manager -- Has[1] --> object_management 
        object_management <--> event_sys 
        object_management --Has[?]--> factory 
        subgraph Bootstrap
            config_reader
        end
        Bootstrap --Initialize--> object_management
        Bootstrap --Feed--> factory
        Bootstrap --Initialize--> event_sys
        Bootstrap --Initialize--> scene_manager
    end

    scene[scene]
    object[object]
    components[components]
    sprites[sprites]

    UI_text[UI text]
    UI_buttons[UI buttons]
    CppMonoBehaviours[CppMonoBehaviours]
    rigid_body[rigid body]
    scripts[scripts]
    animated_sprites[animated sprites]


    Game_engine --Contain[?]--> scene 
    scene --Contain[N]--> object
    object --Contain[N]--> components
    components -.-> sprites
    subgraph Components
        sprites -.-> animated_sprites
        components -.-> UI_text
        components -.-> UI_buttons
        components -.-> CppMonoBehaviours
        components -.-> scripts
        components -.-> rigid_body
    end
    scripts --Can access--> Game_engine
    CppMonoBehaviours --Can access--> Game_engine

```
  Source: [GameDev Stack Exchange](https://gamedev.stackexchange.com/questions/200076/i-dont-get-why-ecs-is-considered-more-performant-than-oop)
```

**Review by 6im0n — CHANGES_REQUESTED** [review]

**Review by Marius-P1 — APPROVED** [review]
:ok_hand:

**Review by axel-eck — APPROVED** [review]

**Review by 6im0n — APPROVED** [review]
Good for me !

**Pull request merged**